src/core/lib/iomgr/tcp_windows.cc

```diff
@@ -482,7 +482,10 @@ static grpc_resource_user* win_get_resource_user(grpc_endpoint* ep) {
   return tcp->resource_user;
 }
 
-static int win_get_fd(grpc_endpoint* ep) { return -1; }
+static int win_get_fd(grpc_endpoint* ep) {
+  grpc_tcp* tcp = reinterpret_cast<grpc_tcp*>(ep);
+  return tcp->socket->socket;
```

Comment:
Could you please use `grpc_winsocket_wrapped_socket()` to get `SOCKET`?

Reply:
@yihuazhang  note that as a maintainer you can apply a suggestion to a PR yourself, you don't need to wait for the original author.
(if not, feel free to provide the snippet and I can add it).